### PR TITLE
Support synchronous delete

### PR DESCRIPTION
**Pull request by oyvindwe**

### CODE REVIEW

File: src/core/Query.java [hunk 1]
```diff
@@ -80,7 +80,36 @@
...
+  /**
+   * Returns whether or not to wait for deletion.
+   * @return A boolean
+   * @since 2.5
+   */
+  boolean getDeleteSync();
+
+  /**
+   * How long to wait for deletion in case {@link #setDeleteSync} is set to true.
+   * Default to 0 (forever).
```
Comment (bnordli) [inline src/core/Query.java]: "Defaults"?

File: src/core/TSQuery.java [hunk 2]
```diff
@@ -95,7 +95,13 @@
   
   /** Whether or not to delete the queried data */
   private boolean delete = false;
-  
+
+  /** Whether or not to wait for the deletion to be completed. */
```
Comment (bnordli) [inline src/core/TSQuery.java:99]: `@since 2.2`?

Comment (oyvindwe) [reply src/core/TSQuery.java:99]: This is added after 2.4. I tried to follow the pattern for `@since`, assuming this will be accepted for 2.5.

**Review by bnordli — APPROVED** [review]

File: src/core/SaltScanner.java [hunk 3]
```diff
@@ -186,8 +186,8 @@ public SaltScanner(final TSDB tsdb, final byte[] metric,
    * combined.
    * @param max_data_points The maximum number of data points pulled out from all
    * scanners (estimated).
-   * @param delete_sync
-   * @param delete_delete_timeout
+   * @param delete_sync Wether or not to wait for deletion
```
Comment (bnordli) [inline src/core/SaltScanner.java]: Whether.

File: src/core/SaltScanner.java [hunk 4]
```diff
@@ -186,8 +186,8 @@ public SaltScanner(final TSDB tsdb, final byte[] metric,
...
+   * @param delete_delete_timeout Milliseconds to wait for deletion. 0 means forever.
```
Comment (bnordli) [inline src/core/SaltScanner.java]: what about only `delete_timeout`?

File: src/core/SaltScanner.java (hunk 3 above)
Comment (oyvindwe) [reply src/core/SaltScanner.java]: Thanks. Fixed.

File: src/core/SaltScanner.java (hunk 4 above)
Comment (oyvindwe) [reply src/core/SaltScanner.java]: Thanks. Fixed.

**Review by bnordli — APPROVED** [review]

**Pull request merged**